pkg/controller/installation/validation.go

```diff
@@ -18,13 +18,19 @@ import (
 	"fmt"
 	"net"
 	"path"
+	"regexp"
 	"strings"
 
 	operatorv1 "github.com/tigera/operator/api/v1"
 	"github.com/tigera/operator/pkg/render"
 	appsv1 "k8s.io/api/apps/v1"
 )
 
+var (
```

Comment:
I think you actually want to just import the Kubernetes validation logic for these. For example:

- For label values: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L167
- For label+annotation keys: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L188

Annotation values are free-form and not structured so I don't think there's any validation we should perform on them.

Reply:
For the label key, this seems to be the right one: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L43